Revert "tests: run `Misc/stats_dir_profiler.swift` only on macos and linux"

compnerd opened the pull request:
Reverts swiftlang/swift#80816 - the justification doesn't explain the issue (simply disabling any long running tests reduces test coverage but doesn't ensure functionality). This is a particularly important thing for things related to paths as those are particularly different across platforms.

compnerd committed:
Revert "tests: run `Misc/stats_dir_profiler.swift` only on macos and linux"

compnerd commented:
@swift-ci please smoke test

eeckstein reviewed (changes requested):
Sorry, we should not re-enable the test as long as we don't know why it takes so long on windows.
My explanation was wrong - I misunderstood an existing comment in the test file. So I don't know what's the problem on windows

xedin commented:
I don't know much about Windows unfortunately but it sounds to me like all of the system calls that stats collector does are more expensive on Windows than on other platforms.

eeckstein commented:
For context, this is the time it took:
```
Slowest Tests:
--------------------------------------------------------------------------
1117.96s: Swift(windows-x86_64) :: Misc/stats_dir_profiler.swift
```

compnerd commented:
> Sorry, we should not re-enable the test as long as we don't know why it takes so long on windows. My explanation was wrong - I misunderstood an existing comment in the test file. So I don't know what's the problem on windows

I don't think that simply disabling the test because it is slow or doesn't work is the right approach. What if I were to introduce a test that works on Windows but fails on macOS? Would it be acceptable to simply disable the test on macOS?

eeckstein commented:
It is not ideal, but in this case I think it's the right compromise, because
1. The test is not testing an end-user feature of the compiler. It's only testing an internal compiler development feature
2. The functionality is still tested on macos and linux. It's unlikely that it only breaks on windows
3. Disabling the test improves PR testing time for _all_ (= many) users.

compnerd commented:
> It is not ideal, but in this case I think it's the right compromise, because
> 
> 1. The test is not testing an end-user feature of the compiler. It's only testing an internal compiler development feature
> 2. The functionality is still tested on macos and linux. It's unlikely that it only breaks on windows
> 3. Disabling the test improves PR testing time for _all_ (= many) users.

One could argue that the same holds for disabling a test that fails on macOS: it still would be tested on Windows. There are platform specific differences, and this potentially is highlighting that there may be one here too. I don't think that this is the right tradeoff unless you are actively looking into it. If you are saying, disabling while you investigate and work to re-enable it, that is a very different thing from, "lets disable until someone someday looks into this".

compnerd commented:
> I don't know much about Windows unfortunately but it sounds to me like all of the system calls that stats collector does are more expensive on Windows than on other platforms.

Perhaps, but at the same time the stats collector is based on LLVM's stats collector which AFAIK works fine on Windows. I think that this might be something with the way it is being used.

xedin commented:
Could you elaborate on what do you mean by fine? It does work it's just slower, our use is not exotic in any way, just dumping things into files like LLVM does.

compnerd commented:
> Could you elaborate on what do you mean by fine? It does work it's just slower, our use is not exotic in any way, just dumping things into files like LLVM does.

What I mean by "fine" is that it is not "prohibitively slow". It might be a small amount slower than on Linux, but there will always be some platform specific differences. If it is similar to what LLVM does, I suspect that there is something else that is the cause of it unless something has broken in LLVM and no one noticed.

compnerd commented:
I think that a better compromise might be to figure out if we can possibly disable the profile-events part of the test or the profile-events part of the test on all platforms to get a more reasonable execution time.

compnerd commented:
Okay, based on a quick test, I think that if we disable the entities profiling, we should shave 14m off of the 18m run. And since the argument was that this is generic infrastructure, I think that limiting it to just one of the two cases is a reasonable tradeoff.

xedin commented:
Sounds good to me!

compnerd committed:
tests: disable the entities profiling

Disable the entities profiling for the stats_dir_profiler as this shaves off ~80% of the test time.

compnerd commented:
@swift-ci please smoke etst

compnerd commented:
@swift-ci please smoke test

compnerd commented:
@swift-ci please smoke test Linux platform

compnerd commented:
Okay, with this change the test is enabled and goes from ~1800s -> ~300s

eeckstein reviewed (approved):
thanks

compnerd commented:
@swift-ci please smoke test Linux platform
